src/platform/plugins/shared/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -248,14 +254,9 @@ class SearchBarUI<QT extends (Query | AggregateQuery) | Query = Query> extends C
   /*
    Keep the "draft" value in local state until the user actually submits the query. There are a couple advantages:
 
-    1. Each app doesn't have to maintain its own "draft" value if it wants to put off updating the query in app state
-    until the user manually submits their changes. Most apps have watches on the query value in app state so we don't
-    want to trigger those on every keypress. Also, some apps (e.g. dashboard) already juggle multiple query values,
-    each with slightly different semantics and I'd rather not add yet another variable to the mix.
-
-    2. Changes to the local component state won't trigger an Angular digest cycle. Triggering digest cycles on every
```

Comment:
a bit of de-angularization :)

Reply:
also this comment explains what happens, but generally this little fix, was very hard to get to. We have so many layers in the Discover - UnifiedSearch - ESQL Editor case, which made it really really hard to get to the right solution. Both me and @stratoula  had several attempts to fix it. On the hindsight, simple change, but due to the complexity of the given use case, very hard to investigate. Nothing to do for now, but longer down the road: Simplification would be great, but of course it's also a matter of time. As the given comment demonstrates, the main code here was written when Kibana still was written in Angular

Reply:
Evidence that every shortcut has a tradeoff 😉